src/code-components/FileInput/FileInput.tsx

```diff
@@ -0,0 +1,97 @@
+import { ChangeEvent, ReactNode, useRef } from "react";
+import { FILE_TYPES } from "./FileInput.register";
+
+interface FileInputProps {
+  onChange?(files: File[]): void;
+  types?: string[];
+  customTypes?: string[];
+  multiple?: boolean;
+  maxSize?: number;
+  className?: string;
+  children: ReactNode;
+}
+
+export function FileInput({
+  onChange,
```

Comment:
Maybe we could have some sort of callback here?

```suggestion
export function FileInput({
  onChange,
  onInvalidFile,
```